src/m365/spp/commands/model/model-remove.spec.ts

```diff
@@ -0,0 +1,191 @@
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation when siteUrl is not valid', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'invalidUrl', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when id is not valid', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com/sites/sales', id: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('correctly handles an error when the model title is not found', async () => {
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        throw {
+          error: {
+            "odata.error": {
+              code: "-1, Microsoft.Office.Server.ContentCenter.ModelNotFoundException",
+              message: {
+                lang: "en-US",
+                value: "File Not Found."
+              }
+            }
+          }
+        };
+      }
+    });
+
+    await assert.rejects(command.action(logger, { options: { verbose: true, siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f', force: true } }),
+      new CommandError('File Not Found.'));
+  });
+
+  it('is the confirmation prompt called', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f' } });
+    assert(confirmationStub.calledOnce);
+  });
+
+  it('deletes model by id', async () => {
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f', force: true } });
+    assert(stubDelete.calledOnce);
+  });
+
+  it('does not delete model when confirmation is not accepted', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
```

Comment:
In this test we don't really care if the API is called correctly, that is validated by other tests. Let's just always resolve the stub without checking for the right API URL.